Source/WebDriver/WebDriverService.cpp

```diff
@@ -325,6 +373,66 @@ void WebDriverService::sendResponse(Function<void (HTTPRequestHandler::Response&
     replyHandler({ result.httpStatusCode(), responseObject->toJSONString().utf8(), "application/json; charset=utf-8"_s });
 }
 
+#if ENABLE(WEBDRIVER_BIDI)
+
+void WebDriverService::handleHandshake(HTTPRequestHandler::Request&& request, Function<void (std::optional<HTTPRequestHandler::Response>&& response)>&& replyHandler)
+{
+    // https://w3c.github.io/webdriver-bidi/#transport
+    // 4.1 Let resource name be the resource name from reading the client’s opening handshake.
+    auto resourceName = request.path;
+
+    // If resource name is not in listener’s list of WebSocket resources, then stop running these steps and act as if the requested service is not available.
+    auto resources = m_bidiServer.listener()->resources;
+    auto foundResource = std::find(resources.begin(), resources.end(), resourceName);
+    if (foundResource == resources.end()) {
+        WTFLogAlways("Resource name %s not found in listener's list of WebSocket resources. Returning 503", resourceName.utf8().data());
+        replyHandler(std::make_optional<HTTPRequestHandler::Response>({ 503, "Service Unavailable", "text/plain"_s }));
+        return;
+    }
+
+    // 4.2 If resource name is the byte string "/session", and the implementation supports BiDi-only sessions:
+    if (*foundResource == "/session"_s) {
```

Comment:
I am not sure that all these spec comments are super helpful, as spec sections will quickly diverge from this and eventually become renamed (and this code is not very complicated to begin with).

Please do keep at least the link at the top of the method, though.

Reply:
Yeah, I had added while implementing the method but now removed most of them, leaving the ones where it might add more context.